### PR TITLE
#158792715 Set up a signup endpoint

### PR DESCRIPTION
**Pull request by elikeyz**
### What does this PR do?
Set up an endpoint for creating a new user account
#### Description of Task to be completed?
- Write a POST request for creating a new user account and generating a JSON Web Token
- Write tests for the endpoint
#### How should this be manually tested?
Test endpoint with Postman and run tests in the command line with "npm test"
#### What are the relevant pivotal tracker stories?
#158792715
#158800700

### CODE REVIEW

**Comment by fob413** [comment]
I can see that your test coverage dropped a lot. Any plans on handling that?

File: controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Pool } from 'pg';
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcryptjs';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const connectString = 'postgres://postgres:mastahacka@localhost:5432/rides';
```
Comment (fob413) [inline controllers/users.js]: Look into having this connection to the database done in a file somewhere and exported out. That way all you have to do in all your controller files is just to import the connected database.

**Review by fob413 — CHANGES_REQUESTED** [review]

File: controllers/users.js (hunk 1 above)
Comment (elikeyz) [reply controllers/users.js]: Ok

File: utils/validate.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const validate = (req, res, next) => {
+  if (!req.body.name) {
+    res.status(400).send('Please enter your name');
+  } else if (!req.body.username) {
+    res.status(400).send('Please enter your username');
+  } else if (!req.body.email) {
+    res.status(400).send('Please enter your email');
+  } else if (!req.body.password) {
+    res.status(400).send('Please enter your password');
```
Comment (fob413) [inline utils/validate.js:9]: you've not tested this yet

**Review by fob413 — CHANGES_REQUESTED** [review]

**Pull request merged**